examples/nodejs/lambda-examples/cloudwatch-metrics/infrastructure/lib/momento-metrics-stack.ts

```diff
@@ -0,0 +1,310 @@
+const customLogGroupName = '/aws/lambda/MomentoMetricsMiddlewareCDKExample';
+
+export class MomentoMetricsStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    const momentoApiKeyParam = new cdk.CfnParameter(this, 'MomentoApiKey', {
+      type: 'String',
+      description: 'The Momento API key that will be used to read from the cache.',
+      noEcho: true,
+    });
+
+    const apiKeySecret = new secrets.Secret(this, 'MomentoMetricsApiKey', {
+      secretName: 'MomentoMetricsApiKey',
+      secretStringValue: new cdk.SecretValue(momentoApiKeyParam.valueAsString),
+    });
+
+    const configToLogGroupName = new Map([
+      [exampleApp.NodejsLambda, '/aws/lambda/MomentoMetricsMiddlewareCDKExample'],
+      [exampleApp.NodejsEcs, '/aws/ecs/MomentoMetricsMiddlewareCDKExample'],
+      [exampleApp.Custom, customLogGroupName]
+    ])
+    const logGroupName = configToLogGroupName.get(stackConfig);
+
+    const logGroup = new LogGroup(this, 'Logs', {
+      logGroupName: logGroupName,
+      retention: RetentionDays.ONE_DAY,
+    });
+
+    switch (stackConfig) {
+      case exampleApp.NodejsLambda: {
+        const nodejsLambda = new lambdaNodejs.NodejsFunction(this, 'MomentoMetricsMiddlewareCDKExample', {
+          functionName: 'MomentoMetricsMiddlewareCDKExample',
+          runtime: lambda.Runtime.NODEJS_16_X,
+          entry: path.join(__dirname, '../../lambda/handler.ts'),
+          projectRoot: path.join(__dirname, '../../lambda'),
+          depsLockFilePath: path.join(__dirname, '../../lambda/package-lock.json'),
+          handler: 'handler',
+          timeout: cdk.Duration.minutes(6),
+          memorySize: 128,
+          environment: {
+            MOMENTO_API_KEY_SECRET_NAME: apiKeySecret.secretName,
+          },
+        });
+        apiKeySecret.grantRead(nodejsLambda);
+        break;
+      }
+      case exampleApp.NodejsEcs: {
+        const cluster = new ecs.Cluster(this, 'MomentoMetricsExampleFargateCluster');
+
+        const imageAsset = new DockerImageAsset(this, "MomentoMetricsECSDockerImage", {
+          directory: path.join(__dirname, "../../docker")
+        });
+
+        const taskDefinition = new ecs.FargateTaskDefinition(this, 'MomentoMetricsMiddlewareCDKExample');
+        taskDefinition.addContainer('MomentoMetricsECSContainer', {
+          image: ecs.ContainerImage.fromDockerImageAsset(imageAsset),
+          logging: new ecs.AwsLogDriver({
+            logGroup: logGroup,
+            streamPrefix: "MomentoMetricsMiddlewareCDKExample"
+          })
+        });
+        taskDefinition.addToTaskRolePolicy(new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            "secretsmanager:ListSecrets",
+            "secretsmanager:GetSecretValue",
+            "logs:PutLogEvents",
+            "logs:CreateLogStream",
+            "ecs:*"
+          ],
+          resources: ["*"],
```

Comment:
nit: we know the secret name for the momento api key above. We should restrict listing/getting secret values to only that secret